src/main/kotlin/com/hj/leetcode/kotlin/problem787/Solution.kt

```diff
@@ -0,0 +1,28 @@
+package com.hj.leetcode.kotlin.problem787
+
+import kotlin.math.min
+
+/**
+ * LeetCode page: [787. Cheapest Flights Within K Stops](https://leetcode.com/problems/cheapest-flights-within-k-stops/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(kM) and Space O(n) where M is the size of flights;
+     */
+    fun findCheapestPrice(n: Int, flights: Array<IntArray>, src: Int, dst: Int, k: Int): Int {
+        // dp[dst]@i::= the minimum price from src to dst in at most i flights;
+        val dp = IntArray(n) { Int.MAX_VALUE }
+        dp[src]= 0
+
+        for (i in 1..k + 1) {
+            val dpPrev = dp.clone()
```

Comment:
Time complexity should be O(kn+kM)

Reply:
Addressed by https://github.com/hj-core/hj-leetcode-kotlin/pull/277/commits/b219c54195723d8c6a44c4c2483e9e588ec85f59